audio/filter/af_lavrresample.c

```diff
@@ -400,10 +400,8 @@ static int control(struct af_instance *af, int cmd, void *arg)
     case AF_CONTROL_RESET:
         if (s->avrctx) {
 #if HAVE_LIBSWRESAMPLE
-            // This POS either can't drop state correctly, or doesn't want to.
-            // It will swallow some minor audio e.g. after a seek.
-            // Deallocate and recreate the resample state for a full reset.
-            configure_lavrr(af, &af->fmt_in, &af->fmt_out, false);
+            swr_close(s->avrctx);
+            swr_init(s->avrctx);
```

Comment:
I don't quite trust it. There's way too much going on in swr_init. Are you absolutely sure it doesn't clobber some input options (which, if swr_init() is run with the clobbered value, would result in different parameters)?

Also swr_init can fail, which you should check. (Call close_lavrr on failure.)